docs/design/proposals/cephfs-snapshot-shallow-ro-vol.md

```diff
@@ -0,0 +1,274 @@
+# Snapshots as shallow read-only volumes
+
+CSI spec doesn't have a notion of "mounting a snapshot". Instead, the idiomatic
+way of accessing snapshot contents is first to create a volume populated with
+snapshot contents and then mount that volume to workloads.
+
+CephFS exposes snapshots as special, read-only directories of a subvolume
+located in `<subvolume>/.snap`. cephfs-csi can already provision writable
+volumes with snapshots as their data source, where snapshot contents are
+cloned to the newly created volume. However, cloning a snapshot to volume
+is a very expensive operation in CephFS as the data needs to be fully copied.
+When the need is to only read snapshot contents, snapshot cloning is extremely
+inefficient and wasteful.
+
+This proposal describes a way for cephfs-csi to expose CephFS snapshots
+as shallow, read-only volumes, without needing to clone the underlying
+snapshot data.
+
+## Use-cases
+
+What's the point of such read-only volumes?
+
+* **Restore snapshots selectively:** users may want to traverse snapshots,
+  restoring data to a writable volume more selectively instead of restoring
+  the whole snapshot.
+* **Volume backup:** users can't backup a live volume, they first need
+  to snapshot it. Once a snapshot is take, it still can't be backed-up,
```

Comment:
is taken?

Reply:
Oops, that's a typo.

Reply:
https://github.com/ceph/ceph-csi/pull/2148/commits/a98a5d30d69c2b25c80edfb413eeb5bd79c7af1a